paramak/parametric_components/port_cutters.py

```diff
@@ -163,3 +170,176 @@ def find_points(self):
+            (-self.width/2, -self.height/2),
+            (self.width/2, -self.height/2),
+            (self.width/2, self.height/2),
+            (-self.width/2, self.height/2),
+        ]
+        points = [(e[0], e[1] + self.z_pos) for e in points]
+        self.points = points
+
+    def add_fillet(self):
+        if self.fillet_radius is not None and self.fillet_radius != 0:
+            self.solid = self.solid.edges('#Z').fillet(self.fillet_radius)
+
+
+class CircularPortCutter(ExtrudeCircleShape):
+    """Creates a extruded shape with a circular section that is used to cut
+    other components (eg. blanket, vessel,..) in order to create ports.
+
+    Args:
+        z_pos (float): Z position (cm) of the port
+        height (float): height (cm) of the port
+        width (float): width (cm) of the port
+        distance (float): extruded distance (cm) of the cutter
+        stp_filename (str, optional): The filename used when saving stp files
+            as part of a reactor. Defaults to "CircularPortCutter.stp".
+        stl_filename (str, optional): The filename used when saving stl files
+            as part of a reactor. Defaults to "CircularPortCutter.stl".
+        color (tuple, optional): (sequences of 3 or 4 floats each in the range
+            0-1): the color to  use when exporting as html graphs or png
+            images. Defaults to (0.5, 0.5, 0.5).
+        azimuth_placement_angle (float or iterable of floats, optional): The
+            angle or angles to use when rotating the shape on the azimuthal
+            axis. Defaults to 0.0.
+        name (str, optional): the legend name used when exporting a html graph
+            of the shape. Defaults to "circular_port_cutter".
+        material_tag (str, optional): The material name to use when exporting
+            the neutronics description. Defaults to "circular_port_cutter_mat".
+    """
+    def __init__(
+        self,
+        z_pos,
+        radius,
+        distance,
+        stp_filename="CircularPortCutter.stp",
+        stl_filename="CircularPortCutter.stl",
+        color=(0.5, 0.5, 0.5),
+        azimuth_placement_angle=0.0,
+        name="circular_port_cutter",
+        material_tag="circular_port_cutter_mat",
+        **kwargs
+    ):
+
+        default_dict = {
+            "points": None,
+            "workplane": "XZ",
+            "solid": None,
+            "intersect": None,
+            "cut": None,
+            "union": None,
+            "tet_mesh": None,
+            "physical_groups": None,
+        }
+
+        for arg in kwargs:
+            if arg in default_dict:
+                default_dict[arg] = kwargs[arg]
+
+        super().__init__(
+            radius=radius,
+            extrude_both=False,
+            name=name,
+            color=color,
+            material_tag=material_tag,
+            stp_filename=stp_filename,
+            stl_filename=stl_filename,
+            azimuth_placement_angle=azimuth_placement_angle - 90,
```

Comment:
@RemDelaporteMathurin should this -90 be here

Reply:
@Shimwell Without this -90, with  azimuth_placement_angle=0 (in the class inititial parameters), the cutter would go straight in the Y direction.

But we expect the azimuth placement angle to be coherent with the rotation angle so 0degree correspond to the X direction. Perhaps a quick call would make it more understandable

Reply:
In fact, what the code does is : 
- extrude a shape from the XZ plane (ie in the Y direction)
- rotate it back to 0degree with the -90 thing

To avoid this -90 thing we could perhaps change the working plane to YZ.